packages/telegram-stamper/src/index.ts

```diff
@@ -0,0 +1,185 @@
+      }
+    }
+  }
+
+  // initialize the telegram stamper by getting/setting the private/public api key values from/to telegram cloud storage
+  async init() {
+    // check to see if were in a telegram mini app context
+    this.checkTelegramContext();
+
+    if (this.stamper) {
+      // insert creds into telegram cloud storage
+      await window.Telegram.WebApp.CloudStorage.setItem(
+        this.getCloudStorageKeyName(),
+        this.stringifyAPIKey(),
+        (err: any, stored: boolean) => {
+          if (err != null || !stored) {
+            throw new TelegramStamperError(
+              `Failed inserting api key into Telegram Cloud Storage${
+                err && `: ${err}`
+              }`
+            );
+          }
+        }
+      );
+    } else {
+      // attempt to get creds from telegram cloud storage
+      await window.Telegram.WebApp.CloudStorage.getItem(
+        this.getCloudStorageKeyName(),
+        (err: any, apiKey: string) => {
+          if (err != null || !apiKey) {
+            throw new TelegramStamperError(
+              `Failed getting api key from Telegram Cloud Storage${
+                err && `: ${err}`
+              }`
+            );
+          }
+
+          const { parsedPublicKey, parsedPrivateKey } =
+            this.parseAPIKey(apiKey);
+
+          // ToDo: api key validation, how does apikeystamper validate creds passed to it?'
+
+          if (!parsedPublicKey || !parsedPrivateKey) {
+            throw new TelegramStamperError(
+              "Failed parsing api key from Telegram Cloud Storage"
+            );
+          }
+
+          // instantiate api key stamper
+          this.stamper = new ApiKeyStamper({
+            apiPublicKey: parsedPublicKey,
+            apiPrivateKey: parsedPrivateKey,
+          });
+        }
+      );
+    }
+  }
+
+  async stamp(payload: string) {
+    // check to see if were in a telegram mini app context
+    this.checkTelegramContext();
+
+    // check to see that the stamper was initialized
+    if (!this.stamper) {
+      throw new TelegramStamperError(
+        "Cannot stamp with unintialized telegram stamper, call TelegramStamper.init()"
+      );
+    }
+
+    return await this.stamper.stamp(payload);
+  }
+
+  // clear key from telegram cloud storage
+  async clearKey(key: string) {
+    // check to see if were in a telegram mini app context
+    this.checkTelegramContext();
+
+    await window.Telegram.WebApp.CloudStorage.removeItem(
+      key,
+      (err: any, removed: boolean) => {
+        if (err || !removed) {
+          throw new TelegramStamperError(
+            `Failed removing key: ${key}${err && `: ${err}`}`
+          );
+        }
+      }
+    );
+  }
+
+  checkTelegramContext() {
+    if (window?.Telegram?.WebApp == null) {
+      throw new TelegramStamperError(
+        "Cannot use telegram stamper in non telegram mini-app environment"
+      );
+    }
+  }
```

Comment:
Instead of checking for `WebApp` let's check for what we need? `Telegram.WebApp.CloudStorage`. I don't know if there's ever a case where `WebApp` would be defined while `WebApp.CloudStorage` isn't, but if there is we should error out right away (could be the case if users have the option to disable cloudstorage someday?)

Reply:
The reason I choose to check for WebApp rather than CloudStorage is there docs: [Once the script is connected, a window.Telegram.WebApp object will become available with the following fields:...](https://core.telegram.org/bots/webapps#initializing-mini-apps), but youre probably right, lets just check for what we need

Reply:
changed to check for window.Telegram.WebApp.CloudStorage